### ISSUE TITLE
Checkout: Region select autofillable even if Allow to Choose State if It is Optional for Country is set to No

### ISSUE THREAD

**Issue by mattijv** [magento/magento2#33437]
### Preconditions (*)

1. Magento 2.4.2 with sample data
2. **United States removed** from _General > General > State Options > **State is Required for**_
3. **Allow to Choose State if It is Optional for Country** is set to **No**

![Screenshot from 2021-07-07 11-33-47](https://user-images.githubusercontent.com/1974547/124733203-a7123a00-df1c-11eb-9f95-a7b32dba8a59.png)

4. **Browser has an address saved** with United States as the country and California as the region (e.g. go to _Settings > Autofill > Addresses and more_ in Chrome and add an address in United States and California as the region). **I only tested this with Chrome**, unsure if other browsers are affected.

![Screenshot from 2021-07-07 11-39-58](https://user-images.githubusercontent.com/1974547/124732698-1d626c80-df1c-11eb-986f-44a31d9b1379.png)

**Note:** the country can be any country with regions, this is just an example.

### Steps to reproduce (*)

1. Add item to the cart
2. Go to the checkout and apply the autofill to the shipping address (in Chrome start typing the name you selected and it should prompt for the autofill).
3. Select shipping method and go to the next step

![Screenshot from 2021-07-07 11-34-33](https://user-images.githubusercontent.com/1974547/124732730-26533e00-df1c-11eb-988f-72928bb903fb.png)

### Expected result (*)

1. The region input is NOT filled.

![Screenshot from 2021-07-07 12-10-13](https://user-images.githubusercontent.com/1974547/124732988-6ca89d00-df1c-11eb-955d-5e0ff4f4b640.png)

### Actual result (*)

1. The region input IS filled.

![Screenshot from 2021-07-07 11-34-51](https://user-images.githubusercontent.com/1974547/124732945-62869e80-df1c-11eb-9f66-372c1fb5ffa8.png)

This is problematic as the **customer can't see the field and does not know if a wrong region is filled** there. We have several orders that have the wrong region in the address as a result of this.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by mattijv** [comment]
@engcom-Hotel Thank you for taking a look, but by your own screenshots you show that you have in fact managed to reproduce the issue: there is no region field visible on the checkout but the region is autofilled to the address. **That is the problem.**

**Comment by konrad-szwed-accenture** [comment]
@magento give me 2.4-develop instance

**Comment by mattijv** [comment]
@engcom-Hotel Any updates on this? Can you look over your results again and confirm that you are able to reproduce the issue, like your screenshots show?

**Comment by engcom-Hotel** [comment]
Hello @mattijv,

I think there might be some confusion. As per my screenshot, the region with the name "California" is visible. And this is also mentioned in your "Actual Result".

Can you please let us know if we need to further work on this ticket?

Thanks

**Comment by mattijv** [comment]
@engcom-Hotel Yes, unfortunately there seems to be confusion. In bug reports the Expected Result and Actual Result are treated as follows:

1. Expected Result: the correct result that SHOULD happen, but DOES NOT
2. Actual Result: the buggy incorrect result that SHOULD NOT happen, but DOES

Your screenshots show that you do indeed get the same result for Actual Result that I do, which is the incorrect and buggy behavior; **the region _California_ should not appear, because there is no (visible) input for it.**

**Comment by mattijv** [comment]
@engcom-Hotel Any updates? Can you confirm that you see the region in the final address and mark this as reproducible?

**Comment by engcom-Hotel** [comment]
Hello @mattijv,

As per the mentioned steps in the description, the issue has been reproducible. So we are confirming this ticket.

Thanks

**Comment by magento-engcom-team** [comment]
@engcom-Hotel Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Hotel 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-43024` were created

**Issue Available**: @engcom-Hotel, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by Melnychuk-Alexandr** [comment]
@magento I am working on this

**Comment by stephansteiner** [comment]
Hi, any progress on this?

**Comment by engcom-Bravo** [comment]
Hi @mattijv,

Thanks for your reporting and collaboration.

We have verified the issue in Latest 2.4-develop instance and the issue is reproducible.Kindly refer the screenshots.

<img width="846" alt="Screenshot 2024-06-24 at 10 47 44" src="https://github.com/magento/magento2/assets/51680745/33324bf2-6a6c-419a-9b3d-4eecfe8d4db3">



<img width="846" alt="Screenshot 2024-06-24 at 10 47 53" src="https://github.com/magento/magento2/assets/51680745/5fab97db-3429-4828-a5b0-6ba70b8eff08">


We are able to see the Region field.

Hence Confirming the issue.

Thanks.

**Comment by mattijv** [comment]
@engcom-Bravo Thank you for confirming this again, though it was [already confirmed once back in 2021](https://github.com/magento/magento2/issues/33437#issuecomment-895041016).

Hopefully two confirmations is enough to get this in the issue queue.

**Comment by lbajsarowicz** [comment]
This browser-autofill-on-hidden-region-field issue has two separate confirmations (2021, 2024) but no PR; I haven't verified the root cause in region.js or the shipping template. Please re-QA on current 2.4-develop and confirm the exact template/script combination that triggers autofill on the hidden field.
